docs/Developer Documentation/Web Engine Support/index.md

```diff
@@ -1,51 +1,100 @@
+### Launching a Webengine App
+
+In order for a Web Engine application to appear on the SDL app list, the HMI must notify SDL Core about newly installed Web Engine applications by sending a `BasicCommunication.SetAppProperties`. The properties sent in this request will be stored in Core's policy table and the information will stay persistent between ignition cycles. 
+
+!!! note
+Not all parameters in the `AppProperties` struct apply to Web Engine Apps. Web Engine Apps should omit `endpoint` and `authToken`.
+!!!
+
+If the users activated a Web Engine application, the HMI will use information from the manifest.js to launch the entrypoint HTML (our development HMIs' do this by creating an invisible iframe). Here, the app will begin execution on the head unit and eventually call `RegisterAppInterface`. When the HMI receives an `OnAppRegistered` notification signalling that the web engine app has successfully registered, the HMI should then send Core an `ActivateApp` request.
+
+## WebSocket Server Transport
+
+In order to support the Webengine feature, a websocket server transport was added to SDL Core. This contrasts to the websocket client transport in SDL Core that is used by Java Cloud applications. When the HMI launches a web engine application, it will provide Core's hostname and port as query parameters to the entrypoint of the web engine application. This transport supports both secure and non-secure websocket communication; which is also determined by a query parameter passed to the entrypoint html file of the web engine application.
+
+These are the accepted values for the sdl-transport-role parameter:
+- ws-server
+- ws-client
+- wss-server
+- wss-client
+- tcp-server
+- tcp-client
 
-### WebSocket Server Transport
+Example URL with query parameters: file://somewhere/HelloSDL/index.html?sdl-host=localhost&sdl-port=12345&sdl-transport-role=wss-server
 
-Before proposal 0240, SDL Core supported websocket transport with the application as the websocket server. With the addition of web engine applications, Core needed to support websocket transport with Core as the websocket server. This change will shift the burden of knowing where the server is from Core to the applications. When the HMI launches one of these web engine applications, it will provide Core's hostname and port as query parameters to the entrypoint of the web engine application. This transport supports both secure and non-secure websocket communication; which is used is also determined by a query parameter passed to the entrypoint html file of the web engine application.
+#### Secured Websocket Connections
 
 Websocket server transport will only run if either all three of these are valid or if none are provided:
 - WSServerCertificatePath (path to websocket server certificate)
```

Comment:
I believe there needs to be a newline above a list for the HTML generator to work